scaleway_functions_python/local/serving.py

```diff
@@ -144,17 +144,57 @@ def resp_record_to_flask_response(
         return resp
 
 
-def _create_flask_app(handler: "hints.Handler") -> Flask:
-    app = Flask(f"serverless_local_{handler.__name__}")
+class LocalFunctionServer:
+    """LocalFunctionServer serves Scaleway FaaS handlers on a local http server."""
 
-    # Create the view from the handler
-    view = HandlerWrapper(handler).as_view(handler.__name__, handler)
+    def __init__(self) -> None:
+        self.app = Flask("serverless_local")
 
-    # By default, methods contains ["GET", "HEAD", "OPTIONS"]
-    app.add_url_rule("/<path:path>", methods=HTTP_METHODS, view_func=view)
-    app.add_url_rule("/", methods=HTTP_METHODS, defaults={"path": ""}, view_func=view)
+    def add_handler(
+        self,
+        handler: "hints.Handler",
+        relative_url: Optional[str] = None,
+        http_methods: Optional[List[str]] = None,
+    ) -> "LocalFunctionServer":
```

Comment:
Why is `LocalFunctionServer` in quotes here? (This might just be my lack of typing knowledge)

Reply:
This is because it's within the class so it has not yet been defined so LocalFunctionServer refers to an unknow identifier in this context. In Python 3.11+, they introduced Self as a way to resolve this https://docs.python.org/3/whatsnew/3.11.html#whatsnew311-pep673 (and also avoid repetition)

With that function, idk if returning self makes sense, it was to make it possible to do `LocalFunctionServer().add_handler(toto).serve()` like with a builder pattern